openapi/powertoolsdeveloper.collections/client.bal

```diff
@@ -214,30 +214,954 @@ public isolated client class Client {
     }
     # Collections - Collection to XML
     #
-    # + payload - Collection conversion parameters
-    # + return - OK
+    # + payload - Collection conversion parameters 
+    # + return - OK 
     remote isolated function collectionToXml(InputCollectionConversionXML payload) returns OutputString|error {
         string  path = string `/CollectionToXML`;
-        map<any> headerValues = {"X-IBM-Client-Id": self.apiKeys["X-IBM-Client-Id"]};
+        map<any> headerValues = {"X-IBM-Client-Id": self.apiKeyConfig.xIbmClientId};
         map<string|string[]> accHeaders = getMapForHeaders(headerValues);
         http:Request request = new;
         json jsonBody = check payload.cloneWithType(json);
         request.setPayload(jsonBody);
         OutputString response = check self.clientEp->post(path, request, headers = accHeaders, targetType=OutputString);
         return response;
     }
-}
-
-# Generate header map for given header values.
-#
-# + headerParam - Headers  map
-# + return - Returns generated map or error at failure of client initialization
-isolated function  getMapForHeaders(map<any> headerParam)  returns  map<string|string[]> {
-    map<string|string[]> headerMap = {};
-    foreach  var [key, value] in  headerParam.entries() {
-        if  value  is  string ||  value  is  string[] {
-            headerMap[key] = value;
-        }
-    }
-    return headerMap;
+    # Currency - Convert currency
+    #
+    # + payload - Currency conversion parameters 
+    # + return - OK 
+    remote isolated function convertCurrency(InputCurrencyConversion payload) returns OutputNumber|error {
```

Comment:
All the power tools connectors have been generated filtering by tags. As an example this connector should be generated using `bal openapi -i openapi.yaml --mode client --tags Collections`. Extra operations added are not under the tag `Collections`. Please check all powertools connectors.